lib/sisimai/bite/email/google.rb

```diff
@@ -0,0 +1,320 @@
+
+        # Google Mail
+        # From: Mail Delivery Subsystem <mailer-daemon@googlemail.com>
+        # Received: from vw-in-f109.1e100.net [74.125.113.109] by ...
+        #
+        # * Check the body part
+        #   This is an automatically generated Delivery Status Notification
+        #   Delivery to the following recipient failed permanently:
+        #
+        #        recipient-address-here@example.jp
+        #
+        #   Technical details of permanent failure:
+        #   Google tried to deliver your message, but it was rejected by the
+        #   recipient domain. We recommend contacting the other email provider
+        #   for further information about the cause of this error. The error
+        #   that the other server returned was:
+        #   550 550 <recipient-address-heare@example.jp>: User unknown (state 14).
+        #
+        #   -- OR --
+        #   THIS IS A WARNING MESSAGE ONLY.
+        #
+        #   YOU DO NOT NEED TO RESEND YOUR MESSAGE.
+        #
+        #   Delivery to the following recipient has been delayed:
+        #
+        #        mailboxfull@example.jp
+        #
+        #   Message will be retried for 2 more day(s)
+        #
+        #   Technical details of temporary failure:
+        #   Google tried to deliver your message, but it was rejected by the recipient
+        #   domain. We recommend contacting the other email provider for further infor-
+        #   mation about the cause of this error. The error that the other server re-
+        #   turned was: 450 450 4.2.2 <mailboxfull@example.jp>... Mailbox Full (state 14).
+        #
+        #   -- OR --
+        #
+        #   Delivery to the following recipient failed permanently:
+        #
+        #        userunknown@example.jp
+        #
+        #   Technical details of permanent failure:=20
+        #   Google tried to deliver your message, but it was rejected by the server for=
+        #    the recipient domain example.jp by mx.example.jp. [192.0.2.59].
+        #
+        #   The error that the other server returned was:
+        #   550 5.1.1 <userunknown@example.jp>... User Unknown
+        #
+        return nil unless mhead['from']    =~ Re0[:from]
+        return nil unless mhead['subject'] =~ Re0[:subject]
+
+        require 'sisimai/address'
+        dscontents = [Sisimai::Bite.DELIVERYSTATUS]
+        hasdivided = mbody.split("\n")
+        rfc822list = []     # (Array) Each line in message/rfc822 part string
+        blanklines = 0      # (Integer) The number of blank lines
+        readcursor = 0      # (Integer) Points the current cursor position
+        recipients = 0      # (Integer) The number of 'Final-Recipient' header
+        statecode0 = 0      # (Integer) The value of (state *) in the error message
+        v = nil
+
+        hasdivided.each do |e|
+          if readcursor.zero?
+            # Beginning of the bounce message or delivery status part
+            if e =~ Re1[:begin]
```

Comment:
__[rubocop]__
* Style/IfUnlessModifier: Favor modifier `if` usage when having a single-line body. Another good alternative is the usage of control flow `&&`/`||`. [:ref](https://github.com/bbatsov/ruby-style-guide#if-as-a-modifier)

<!-- Comment by [SideCI](https://sideci.com) -->